kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/CreateGenerator.kt

```diff
@@ -0,0 +1,80 @@
+package me.tatarka.inject.compiler
+
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.plusParameter
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import kotlin.reflect.KClass
+
+class CreateGenerator(private val astProvider: AstProvider, private val options: Options) {
```

Comment:
I'm a bit skeptical that this needs to be broken up into it's own class. And if so, it should be an implementation detail of `InjectGenerator`, instead of being passed as an argument.

Reply:
I think this class nicely groups together the code for generating a components create method without the need for functions defined in functions. I didn't want to do too many changes at once in this pr but I also think it would be possible to remove the Options flag from this class and instead have several impls of a `CreateGenerator` interface `CompanionCreateGenerator` & `ClassCreateGenerator`.